### ISSUE TITLE
[no-explicit-any] ignores generic in function definition

### ISSUE THREAD

**Issue by MayhemYDG** [typescript-eslint/typescript-eslint#64]
**Repro**

```TS
function test<T extends Partial<any>>() {}

const test = <T extends Partial<any>>() => {};
```

**Expected Result**
`any` should not be allowed there with the rule `no-explicit-any` enabled.
**Actual Result**
`any` is not detected.

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `0.14.0` |
| `typescript-eslint-parser` | `21.0.2` |
| `typescript`               | `3.2.2` |

**Comment by bradzacher** [comment]
Should have a test case added to the plugin if this is fixed.
